dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt

```diff
@@ -6,49 +6,148 @@
+import com.datadog.android.log.internal.domain.event.LogEventSerializer
 import com.datadog.android.log.model.LogEvent
-import com.datadog.android.v2.core.internal.storage.Storage
+import com.datadog.android.v2.api.FeatureEventReceiver
+import com.datadog.android.v2.api.SdkCore
+import com.datadog.android.v2.core.internal.storage.DataWriter
+import com.datadog.android.v2.core.internal.storage.NoOpDataWriter
+import com.datadog.android.v2.log.internal.storage.LogsDataWriter
+import java.util.Locale
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
 import java.util.concurrent.atomic.AtomicBoolean
 
 internal class LogsFeature(
-    private val coreFeature: CoreFeature,
-    private val storage: Storage
-) {
-    internal var persistenceStrategy: PersistenceStrategy<LogEvent> = NoOpPersistenceStrategy()
+    private val sdkCore: SdkCore
+) : FeatureEventReceiver {
+
+    internal var dataWriter: DataWriter<LogEvent> = NoOpDataWriter()
     internal val initialized = AtomicBoolean(false)
+    private val logGenerator = DatadogLogGenerator()
 
     internal fun initialize(configuration: Configuration.Feature.Logs) {
-        persistenceStrategy = createPersistenceStrategy(storage, configuration)
+        sdkCore.setEventReceiver(LOGS_FEATURE_NAME, this)
+        dataWriter = createDataWriter(configuration)
         initialized.set(true)
     }
 
     internal fun stop() {
-        persistenceStrategy = NoOpPersistenceStrategy()
+        sdkCore.removeEventReceiver(LOGS_FEATURE_NAME)
+        dataWriter = NoOpDataWriter()
         initialized.set(false)
     }
 
-    // region SdkFeature
+    // region FeatureEventReceiver
+
+    @AnyThread
+    override fun onReceive(event: Any) {
+        if (event !is Map<*, *>) {
+            devLogger.w(UNSUPPORTED_EVENT_TYPE.format(Locale.US, event::class.java.canonicalName))
+            return
+        }
+
+        if (event["type"] == "crash") {
+            sendCrashLog(event)
+        } else {
+            devLogger.w(UNKNOWN_EVENT_TYPE_PROPERTY_VALUE.format(Locale.US, event["type"]))
+        }
+    }
+
+    // endregion
+
+    // region Internal
 
-    private fun createPersistenceStrategy(
-        storage: Storage,
+    private fun createDataWriter(
         configuration: Configuration.Feature.Logs
-    ): PersistenceStrategy<LogEvent> {
-        return LogFilePersistenceStrategy(
-            coreFeature.contextProvider,
-            coreFeature.persistenceExecutorService,
-            configuration.logsEventMapper,
-            storage
+    ): DataWriter<LogEvent> {
+        return LogsDataWriter(
+            serializer = MapperSerializer(
+                LogEventMapperWrapper(configuration.logsEventMapper),
+                LogEventSerializer()
+            ),
+            internalLogger = sdkLogger
         )
     }
 
+    @Suppress("ComplexMethod")
+    private fun sendCrashLog(data: Map<*, *>) {
+        val threadName = data["threadName"] as? String
```

Comment:
They are not in the constants on purpose, because caller (crash reporting/NDK crash reporting) and consumer (logs) will be in the different modules without any dependency. We have a task in the backlog to document messages format though.

Maybe later will create some common constants in the core module, will see.